docs/faq.md

```diff
@@ -38,29 +36,73 @@ hash after their name, then this unexpected. Check to see if your build is
 adding multiple configurations of the same targets to the build graph. If you
 need help, reach out to us.
 
-## My Xcode project seems to be of of sync with my Bazel project. What should I do?
+## What is `CompileStub.m`?
+
+If you have a top level target, such as `ios_application`, and its primary
+library dependency is also directly depended on by another top level target,
+such as `ios_unit_test`, then we can't merge that library into the first top
+level target. When that happens, the first top level target doesn't have any
+source files, so we need to add a stub one to allow Xcode to link to the proper
+library target.
+
+If this setup isn't desired (e.g. wanting to have the target merged to enable
+SwiftUI Previews), there are a couple ways to fix it. For tests, setting the
+first top level target as the `test_host` will allow for the library to merge.
+In other cases, refactor the build graph to have the shared code in its own
+library separate from the top level target's primary library.
+
+## Why do some of my `swift_library`s compile twice in BwX mode?
+
+If you have header based rules that depend on the Swift generated header (i.e.
+custom modulemap or hmap rules), then when BwX mode generates those files with
+Bazel, it causes Bazel to generate the Swift generated header by compiling the
+swiftmodule.
 
-The generated Xcode project includes scripts to synchronize select Bazel
-generated files (e.g. `Info.plist`) with Xcode. Perform the following steps to
-synchronize these file:
+Possible solutions are:
 
-1. Open the Xcode project: `xed path/to/MyApp.xcodeproj`.
-2. Select the `Bazel Generated Files` scheme (Menu: `Product` > `Scheme` >
-   `Bazel Generated Files`).
-3. Build the `Bazel Generated Files` scheme (Menu: `Product` > `Build`).
-4. If items under the `Bazel Generated Files` group in the Project navigator are
-   red, close and re-open the project.
+- Refactoring mixed-language targets to single language targets, removing the
+  need for the custom modulemap that references the Swift generated header
+- Remove the use of header maps (hmaps), or don't include Swift generated
+  headers in them
+- Use BwB mode instead
 
-All targets that depend on generates files depend on the `Bazel Generated Files`
-target, so building any of those targets will also synchronize Xcode.
+## Why does Xcode complain about stale files?
 
-## When I build I get warnings like "Stale file 'PROJECT.xcodeproj/rules_xcodeproj/gen_dir/...' is located outside of the allowed root paths". How do I fix this?
+The generated project tracks Bazel managed external and generated files. If you
```

Comment:
> The generated project tracks Bazel managed external and generated files.

I just realize that BwB mode probably doesn't need to do this anymore. I'll add a TODO to explore this.